### ISSUE TITLE
Problems while hashing Big Numbers

### ISSUE THREAD

**Issue by dhruvkelawala** [paulmillr/micro-starknet#2]
Hashing fails while trying run `pedersen` on big numbers even if they are smaller than Field Prime. Not sure how calculations are done but [this](https://github.com/paulmillr/micro-starknet/blob/0a3d722a2237618f4fffa62ee948ac0d511dde36/index.ts#L892) function seems responsible for it which is called by [normalizeScalar](https://github.com/paulmillr/micro-starknet/blob/0a3d722a2237618f4fffa62ee948ac0d511dde36/index.ts#L696) function.

I am getting the following TypeError: `Expected valid private scalar: 0 < scalar < curve.n, got:  3618502788666131213697322783095070105623107215331596699973092056135872020475: bigint`. This is from the **updated** error message from `normaliseScalar`.  (Updated locally)

```typescript
function normalizeScalar(num: number | bigint): bigint {
  if (typeof num === 'number' && Number.isSafeInteger(num) && num > 0) return BigInt(num);
  if (typeof num === 'bigint' && isWithinCurveOrder(num)) return num;
  throw new TypeError(
    `Expected valid private scalar: 0 < scalar < curve.n, got:  ${num}: ${typeof num}`
  );
}
```

The number showed in the error is less than Prime.
These numbers that I am trying to hash passes with old implementation of pedersen in starknet.js and rust-wasm.

**Comment by dhruvkelawala** [comment]
This is the bytecode of data that I am trying to hash with `computeHashOnElements`

Data: https://gist.github.com/dhruvkelawala/6ee172f1011c4b19ddaa50f758a4cf9e

```typescript
export function computeHashOnElements(data: BigNumberish[]) {
  const hexData = data.map((d) => toHex(toBN(d)));

  return [...hexData, hexData.length]
    .reduce((x: PedersenArg, y: PedersenArg) => addHexPrefix(pedersen(x, y)), 0)
    .toString();
}
```

**Comment by paulmillr** [comment]
fixed in 074f418
